File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/Chamodev-max.py
"""/*
 * EJERCICIO:
 * - Crea ejemplos utilizando todos los tipos de operadores de tu lenguaje:
 *   Aritméticos, lógicos, de comparación, asignación, identidad, pertenencia, bits...
 *   (Ten en cuenta que cada lenguaje puede poseer unos diferentes)
 * - Utilizando las operaciones con operadores que tú quieras, crea ejemplos
 *   que representen todos los tipos de estructuras de control que existan
 *   en tu lenguaje:
 *   Condicionales, iterativas, excepciones...
 * - Debes hacer print por consola del resultado de todos los ejemplos.
 *
 * DIFICULTAD EXTRA (opcional):
 * Crea un programa que imprima por consola todos los números comprendidos
 * entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
 *
 * Seguro que al revisar detenidamente las posibilidades has descubierto algo nuevo.
 */"""

a = 5;
b = 2;

#ARITMETICOS
print (a + b); #suma
print (a - b); #reta
print (a * b); #multiplicacion
print (a / b); #division
print (a // b); #division con decimal
print (a % b); #modulo
print (a ** b); #exponente

#ASIGNACION
a += 18; #asignación a suma
print (a);
a -= 18; #asignación a resta
print (a);
a *= 18; #asignación a multiplicación
print (a);
a /= 18; #asignación a división
print (a);
a //= 18; #asignación a division con decimal
print (a);
a %= 18; #asignación a modulo
print (a);
a **= 18; #asignación a exponente
print (a);

#COMPARACION
print("¿10 == 3?", 10 == 3)
print("¿10 != 3?", 10 != 3)
print("¿10 > 3?", 10 > 3)
print("¿10 < 3?", 10 < 3)
print("¿10 >= 10?", 10 >= 10)
print("¿10 <= 9?", 10 <= 9)

#LOGICOS
a = True
b = False

print("a and b:", a and b)
print("a or b:", a or b)
print("not a:", not a)

#IDENTIDAD
x = [1, 2, 3]
y = x
z = [1, 2, 3]

print("x is y:", x is y)  # Mismo objeto
print("x is z:", x is z)  # Contenido igual, pero diferente objeto
print("x is not z:", x is not z)

#PERTENENCIA
lista = [1, 2, 3, 4]

print("2 in lista:", 2 in lista)
print("5 not in lista:", 5 not in lista)

#BITS
a = 5  # 0101
b = 3  # 0011

print("a & b:", a & b)    # 0001
print("a | b:", a | b)    # 0111
print("a ^ b:", a ^ b)    # 0110
print("~a:", ~a)          # Complemento
print("a << 1:", a << 1)  # 1010
print("a >> 1:", a >> 1)  # 0010

# Estrutura básica
# Condicional
edad = 18
if edad < 12:
    print("Crianza")
elif edad < 18:
    print("Adolescente")
else:
    print("Adulto")

#ITERATIVAS
contador = 0
while contador < 3:
    print("Contando:", contador)
    contador += 1

for i in range(3):
    print("Iteración con for:", i)

for i in range(5):
    if i == 3:
        continue
    if i == 4:
        break
    print("for con continue/break:", i)

#MANEJO DE EXCEPCIONES
try:
    resultado = 10 / 0
except ZeroDivisionError:
    print("¡Error! División por cero.")
finally:
    print("Bloque finally ejecutado.")

for i in range(10, 56):
    if (i % 2 == 0) and (i != 16) and (i % 3 != 0):
        print(i)

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/Chamodev-max.js
/*
 * EJERCICIO:
 * - Crea ejemplos utilizando todos los tipos de operadores de tu lenguaje:
 *   Aritméticos, lógicos, de comparación, asignación, identidad, pertenencia, bits...
 *   (Ten en cuenta que cada lenguaje puede poseer unos diferentes)
 * - Utilizando las operaciones con operadores que tú quieras, crea ejemplos
 *   que representen todos los tipos de estructuras de control que existan
 *   en tu lenguaje:
 *   Condicionales, iterativas, excepciones...
 * - Debes hacer print por consola del resultado de todos los ejemplos.
 *
 * DIFICULTAD EXTRA (opcional):
 * Crea un programa que imprima por consola todos los números comprendidos
 * entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
 *
 * Seguro que al revisar detenidamente las posibilidades has descubierto algo nuevo.
 */

console.log("=== OPERADORES ===");

// Aritméticos
let a = 10;
let b = 3;
console.log("Suma:", a + b);
console.log("Resta:", a - b);
console.log("Multiplicación:", a * b);
console.log("División:", a / b);
console.log("Módulo:", a % b);
console.log("Exponente:", a ** b);

// Asignación
let x = 5;
x += 2;
console.log("Asignación +=:", x);

// Comparación
console.log("Mayor que:", a > b);
console.log("Menor o igual:", a <= b);
console.log("Igualdad (==):", a == "10");
console.log("Igualdad estricta (===):", a === "10");

// Lógicos
console.log("AND lógico:", true && false);
console.log("OR lógico:", true || false);
console.log("NOT lógico:", !false);

// Identidad
console.log("Igual estricto:", 5 === "5"); // false
console.log("No igual estricto:", 5 !== "5"); // true

// Pertenencia
let persona = { nombre: "Ana", edad: 25 };
console.log("'nombre' in persona:", "nombre" in persona); // true

let frutas = ["manzana", "banana", "uva"];
console.log("¿Incluye 'uva'?:", frutas.includes("uva")); // true

// Bits
let m = 5; // 0101
let n = 3; // 0011
console.log("AND a nivel de bits:", m & n); // 0001
console.log("OR a nivel de bits:", m | n);  // 0111
console.log("XOR a nivel de bits:", m ^ n); // 0110
console.log("Desplazamiento a la izquierda:", m << 1); // 1010 (10)
console.log("Desplazamiento a la derecha:", m >> 1); // 0010 (2)

console.log("\n=== ESTRUCTURAS DE CONTROL ===");

// Condicional if - else
let edad = 20;
if (edad < 18) {
    console.log("Eres menor de edad.");
} else if (edad >= 18 && edad < 60) {
    console.log("Eres adulto.");
} else {
    console.log("Eres adulto mayor.");
}

// Switch
let dia = "miércoles";
switch (dia) {
    case "lunes":
        console.log("Inicio de semana");
        break;
    case "miércoles":
        console.log("Mitad de semana");
        break;
    case "viernes":
        console.log("Fin de semana cercano");
        break;
    default:
        console.log("Día cualquiera");
}

// Bucle for
for (let i = 1; i <= 3; i++) {
    console.log("for loop:", i);
}

// Bucle while
let contador = 0;
while (contador < 3) {
    console.log("while loop:", contador);
    contador++;
}

// Bucle do...while
let j = 0;
do {
    console.log("do...while loop:", j);
    j++;
} while (j < 3);

// for...of (para recorrer arrays)
for (let fruta of frutas) {
    console.log("Fruta con for...of:", fruta);
}

// for...in (para recorrer propiedades de un objeto)
for (let key in persona) {
    console.log(`Clave: ${key}, Valor: ${persona[key]}`);
}

// Excepciones
try {
    let resultado = 10 / 0;
    if (!isFinite(resultado)) {
        throw new Error("División por cero no permitida");
    }
    console.log("Resultado:", resultado);
} catch (error) {
    console.log("Error capturado:", error.message);
} finally {
    console.log("Bloque finally siempre se ejecuta");
}

for (let i = 10; i < 56; i++){
 if (i % 2 === 0 && i != 16 && i % 3 !== 0){
    console.log(i)
}
}
